### PR TITLE
Added image validation and functional tests.

### PR DESCRIPTION
**Pull request by jtessler**
Added image size and type validation. Added image upload tests. Also updated every functional test, which are now all green.

### CODE REVIEW

**Comment by amshali** [comment]
Looks good. Let me pull it to my local and test it.

File: app/controllers/Photos.java [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-import controllers.Secure;
+import play.db.jpa.*;
 import models.*;
 
+@With(Secure.class)
 public class Photos extends OBController {
 
+  /* All possible image mime types in a single regex. */
+  public static final String IMAGE_TYPE = "^image/(gif|jpeg|pjpeg|png)$";
+  public static final int MAX_FILE_SIZE = 500 * 1024;  /* Max size in bytes. */
```
Comment (amshali) [inline app/controllers/Photos.java:17]: Can we resize the image if its size exceed this amount?

Comment (jtessler) [inline app/controllers/Photos.java:17]: Yes, I think that's something we should add. I created a new feature for this on the spreadsheet.

Comment (amshali) [inline app/controllers/Photos.java:17]: Thank you. It'd be interesting to see what is the algorithm for converting an image to be under a given size.

**Pull request merged**